src/main/java/com/google/sps/servlets/RepListServlet.java

```diff
@@ -5,11 +5,67 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.Scanner;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.util.EntityUtils;
+import org.apache.http.HttpEntity;
+import java.net.URISyntaxException;
+import java.net.URI;
+import org.apache.http.impl.client.HttpClients;
+import com.google.gson.Gson;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import io.github.cdimascio.dotenv.Dotenv;
 
 @WebServlet ("/rep_list")
 public class RepListServlet extends HttpServlet{
+    Dotenv dotenv = Dotenv.load();
+    private final String API_KEY = dotenv.get("CIVIC_API_KEY");
+    private static final Logger logger = LogManager.getLogger("Errors");
+
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
+        String zipcode = request.getParameter("zipcode");
+        HttpClient httpclient = HttpClients.createDefault();
+
+        URIBuilder builder = new URIBuilder();
+        builder.setScheme("https").setHost("www.googleapis.com/civicinfo/v2/representatives")
+        .setParameter("key", API_KEY)
+        .setParameter("address", zipcode);
+        URI uri = null;
+        try{
+            uri = builder.build();
+        } catch(URISyntaxException e){
+            logger.error(e);
```

Comment:
exit program after logging error

Reply:
I don't think exiting is the behavior you want (here and elsewhere). You should return an error to the caller so they know what has happened. doGet() should also be able to throw a ServletException. Maybe the servlet patten does things a little differently, but generally you either you know how to handle the exception and continue on the flow or a different flow. Or you don't know what to do so you throw it and let the caller handle it.

What happens to the rest of the application when calling exit()?